### PR TITLE
Feat/lesq 561/specialist download page [LESQ-561]

### PR DESCRIPTION
**Pull request by njblore**
## Description

Music year: 1974

- adds queries, schema and page for specialist lesson downloads
- next lesson links on downloads success page still TODO

## How to test

Yet another 'no changes in preview' PR. 
1. Go to https://deploy-preview-2306--oak-web-application.netlify.thenational.academy/teachers/specialist/programmes/independent-living-applying-learning/units/staying-safe-al-5556/lessons/online-safety-c5gk8r/downloads
2. Should be 'internal error' (404)
3. You should see the rest of the site remains the same

## Screenshots

How it should now look:

Screenshot 2024-03-11 at 11 50 00

### CODE REVIEW

File: src/pages/teachers/specialist/programmes/[programmeSlug]/units/[unitSlug]/lessons/[lessonSlug]/downloads.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+      try {
+        const curriculumData =
+          await curriculumApi2023.specialistLessonDownloads({
+            programmeSlug,
+            unitSlug,
+            lessonSlug,
+          });
+
+        if (!curriculumData) {
```
Comment (k-huggs) [inline src/pages/teachers/specialist/programmes/[programmeSlug]/units/[unitSlug]/lessons/[lessonSlug]/downloads.tsx:77]: Does the try / catch not do the same thing twice, throwing an error if there's no `curriculumData`?

**Review by k-huggs — APPROVED** [review]
Just one question

File: src/pages/teachers/specialist/programmes/[programmeSlug]/units/[unitSlug]/lessons/[lessonSlug]/downloads.tsx (hunk 1 above)
Comment (njblore) [reply src/pages/teachers/specialist/programmes/[programmeSlug]/units/[unitSlug]/lessons/[lessonSlug]/downloads.tsx:77]: the query itself can throw an error so we need to catch that and show the 404 page, whereas this just shows the 404 if the query succeeds but returns undefined/null

Comment (k-huggs) [reply src/pages/teachers/specialist/programmes/[programmeSlug]/units/[unitSlug]/lessons/[lessonSlug]/downloads.tsx:77]: ahh ok ok, I think I presumed that the query file would catch the error and throw it at that point

**Review by jack-skerrett-oak — APPROVED** [review]
Legacy and new downloads still seem to work!

**Pull request merged**